File: lib/markdown.ts
import { chunkSection, splitIntoSections } from "./chunker";

export type Chunk = {
  title: string;
  text: string;
};

export function buildMarkdown(text: string, sourceName: string): string {
  const sections = splitIntoSections(text);
  const chunks: Chunk[] = [];
  for (const section of sections) {
    for (const chunk of chunkSection(section)) {
      chunks.push({ title: section.title, text: chunk });
    }
  }

  const header = buildHeader(sourceName, chunks.length);
  const body = chunks.map(renderChunk).join("\n\n");
  return `${header}\n\n${body}`.trim();
}

function buildHeader(sourceName: string, totalChunks: number): string {
  const timestamp = new Date().toISOString().replace("T", " ").slice(0, 16) + " UTC";
  return [
    "# Documento jurídico convertido",
    "",
    `- Fuente: \`${sourceName}\``,
    `- Fecha de conversión: ${timestamp}`,
    `- Total de chunks: ${totalChunks}`
  ].join("\n");
}

function renderChunk(chunk: Chunk): string {
  return [chunk.title, "", chunk.text, "", "---"].join("\n");
}

File: src/legal_docling/markdown.py
from __future__ import annotations

from dataclasses import dataclass
from datetime import datetime

from .chunker import Section, chunk_section, split_into_sections


@dataclass
class Chunk:
    title: str
    index: int
    text: str


def build_markdown(text: str, source_name: str) -> str:
    sections = split_into_sections(text)
    chunks: list[Chunk] = []
    chunk_index = 1

    for section in sections:
        for chunk in chunk_section(section):
            chunks.append(Chunk(title=section.title, index=chunk_index, text=chunk))
            chunk_index += 1

    header = _build_header(source_name, len(chunks))
    body = "\n\n".join(_render_chunk(chunk) for chunk in chunks)
    return f"{header}\n\n{body}".strip()


def _build_header(source_name: str, total_chunks: int) -> str:
    timestamp = datetime.utcnow().strftime("%Y-%m-%d %H:%M UTC")
    return (
        f"# Documento jurídico convertido\n\n"
        f"- Fuente: `{source_name}`\n"
        f"- Fecha de conversión: {timestamp}\n"
        f"- Total de chunks: {total_chunks}"
    )


def _render_chunk(chunk: Chunk) -> str:
    return (
        f"## Chunk {chunk.index}\n\n"
        f"**Sección:** {chunk.title}\n\n"
        f"{chunk.text}\n\n"
        f"---"
    )
